src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/Dispensers.cs

```diff
@@ -43,6 +43,30 @@ internal static RuntimeAssembly GetRuntimeAssembly(RuntimeAssemblyName assemblyR
         }
 
         /// <summary>
+        /// Returns non-null or throws.
+        /// </summary>
+        internal static RuntimeAssembly GetRuntimeAssemblyFromByteArray(byte[] rawAssembly, byte[] pdbSymbolStore)
+        {
+            AssemblyBinder binder = ReflectionCoreExecution.ExecutionDomain.ReflectionDomainSetup.AssemblyBinder;
+            AssemblyBindResult bindResult;
+            Exception exception;
+            if (!binder.Bind(rawAssembly, pdbSymbolStore, out bindResult, out exception))
+            {
+                if (exception != null)
+                    throw exception;
+                else
+                    throw new BadImageFormatException();
+            }
+
+            RuntimeAssembly result = null;
+            GetEcmaRuntimeAssembly(bindResult, ref result);
```

Comment:
:) Its not technically a formality. Its legal to use Load(byte[]) to specify an assembly which is already loaded, and to find it in some circumstances. Fixed.